CMakeLists.txt

```diff
@@ -69,6 +69,14 @@ check_include_file_cxx ("boost/shared_ptr.hpp" HAVE_BOOST)
 check_library_exists (dl dlopen "" HAVE_LIBDL)
 check_function_exists (popen HAVE_POPEN)
 
+if (MSVC)
+  set (CMAKE_C_FLAGS "/WX ${CMAKE_C_FLAGS}")
+  set (CMAKE_CXX_FLAGS "/EHsc /WX ${CMAKE_CXX_FLAGS}")
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  set (CMAKE_C_FLAGS "-Wall -Wextra ${CMAKE_C_FLAGS}")
+  set (CMAKE_CXX_FLAGS "-Wall -Wextra ${CMAKE_CXX_FLAGS}")
+endif ()
+
```

Comment:
The approach that we use in the autotools and I think is common across open source projects is to **not** enable warnings as errors by default. However, we do turn this on in the Travis/Appveyor tests. The thinking is that you want the code to build with warnings that later versions of the compilers may add without breaking the build for users. However, we want to break the build during development if a warning appears. So in summary, move `/WX` into Appveyor.yml.

Reply:
Yes, I prefer that